### PR TITLE
config extends changed to slice,makes configs in order

### PR DESCRIPTION
**Pull request by nejisama**
extends changed  from map to slice, makes the extend callback can be called in order.

### CODE REVIEW

**Review by taoyuanyuan — APPROVED** [review]
LGTM

File: pkg/configmanager/effectiveconfig.go [hunk 1]
```diff
@@ -132,7 +132,21 @@ func SetRouter(router v2.RouterConfiguration) {
...
 	configLock.Lock()
 	defer configLock.Unlock()
-	conf.ExtendConfigs[typ] = cfg
+	found := false
+	for idx, ext := range conf.ExtendConfigs {
+		if ext.Type == typ {
+			ext.Config = cfg
+			conf.ExtendConfigs[idx] = ext
+			found = true
+			break
```
Comment (cch123) [inline pkg/configmanager/effectiveconfig.go:141]: 这里return的话是不是就不需要判断found了

Comment (cch123) [reply pkg/configmanager/effectiveconfig.go:141]: 不过后面还有个trydump

Comment (nejisama) [reply pkg/configmanager/effectiveconfig.go:141]: 不能return，需要trydump

Comment (cch123) [reply pkg/configmanager/effectiveconfig.go:141]: 理论上可以goto ：）

**Review by cch123 — APPROVED** [review]

**Comment by cch123** [comment]
LGTM

**Pull request merged**